HOTT-3513: Add build job for new AWS environment

amberstarlight opened the pull request:
### Jira link

[HOTT-3513](https://transformuk.atlassian.net/browse/HOTT-3513)

### What?

I have added/removed/altered:

- Added a new CircleCI job to build and push images to the new AWS ECR repository.
- Broken this out into a shell script, which I've checked against `shellcheck`.

### Why?

- Multiple things needed to change here:
  - moving away from using the `dev-build` parameter (i.e. just using the short SHA for the tag), and
  - having to use a different context for the new AWS ECR repository (as it's on a different account).

### Other notes

I would have liked to add a `.pre-commit-config.yml` to this repository, but even running the basic things like end-of-file-fixer made changes to a staggering number of files across the repository. That resulted in a _giant_ diff that would take a significant amount of effort to review.

I think we should make an ADR (and, ostensibly, a new ticket on the backlog) to get this implemented at some stage for consistency across the repo. We could break it up per subdirectory in several smaller PRs, or do it all at once in a co-ordinated strike.

amberstarlight committed:
HOTT-3513: Add build job for new AWS environment

willfish commented on .circleci/config.yml:
```diff
@@ -372,6 +372,25 @@ jobs:
+      environment:
+        type: string
+    docker:
+      - image: cimg/ruby:3.2.2-node
```
I wonder if we could move to a smaller image?

amberstarlight replied:
It has to build the application - `cimg/ruby:3.2.2-node` happens to have all the binaries we need for doing that. Is there any other image we could use to build?

willfish replied:
Ah k

willfish reviewed (dismissed):
LGTM

amberstarlight committed:
HOTT-3513: Update push script

After creating an ECR repo for each application, I have now updated
the script to fetch the new ECR URL, and build the container name using
the correct string.

amberstarlight committed:
HOTT-3513: Tag image correctly in all commands

We have to tag images always, otherwise it defaults to latest.

amberstarlight committed:
HOTT-3513: Print shell commands and return codes

Let's debug!

amberstarlight committed:
HOTT-3513: Fix docker push command

The repository URL that we fetch from SSM already has the name of the
image we want to push, so we can, simply, append the docker tag and
push!

amberstarlight committed:
HOTT-3513: Correct the tag of the image we built

willfish reviewed (approved):
Feel free to ignore the test failures. They are totally unrelated to your change. I'll fix these separately
